src/main/java/com/google/sps/placeGuide/repository/impl/DatastorePlaceGuideRepository.java

```diff
@@ -0,0 +1,124 @@
+
+    return placeGuideEntity;
+  }
+
+  @Override
+  public List<PlaceGuide> getAllPublicPlaceGuides() {
+    Filter queryFilter = new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, true);
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Override
+  public List<PlaceGuide> getCreatedPlaceGuides(String creatorId) {
+    Filter queryFilter = new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId);
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Override
+  public List<PlaceGuide> getCreatedPublicPlaceGuides(String creatorId) {
+    Filter queryFilter = CompositeFilterOperator.and(Arrays.asList(
+                        new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId),
+                        new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, true)));
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  @Override
+  public List<PlaceGuide> getCreatedPrivatePlaceGuides(String creatorId) {
+    Filter queryFilter = CompositeFilterOperator.and(Arrays.asList(
+                        new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId),
+                        new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, false)));
+    Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
+    return getPlaceGuidesList(query);
+  }
+
+  private List<PlaceGuide> getPlaceGuidesList(Query query) {
+    PreparedQuery results = datastore.prepare(query);
+    List<PlaceGuide> createdPlaceGuides = new ArrayList<>();
+    for (Entity placeGuideEntity : results.asIterable()) {
+      PlaceGuide placeGuide = getPlaceGuideFromEntity(placeGuideEntity);
+      createdPlaceGuides.add(placeGuide);
+    }
+    return Collections.unmodifiableList(createdPlaceGuides);
+  }
+
+  private PlaceGuide getPlaceGuideFromEntity(Entity placeGuideEntity) {
+    long id = placeGuideEntity.getKey().getId();
+    String name = (String) placeGuideEntity.getProperty(NAME_PROPERTY);
+    String audioKey = (String) placeGuideEntity.getProperty(AUDIO_KEY_PROPERTY);
+    String creatorId = (String) placeGuideEntity.getProperty(CREATOR_ID_PROPERTY);
+    String placeId = (String) placeGuideEntity.getProperty(PLACE_ID_PROPERTY);
+    GeoPt coordinate = (GeoPt) placeGuideEntity.getProperty(COORDINATE_PROPERTY);
+    boolean isPublic = (boolean) placeGuideEntity.getProperty(IS_PUBLIC_PROPERTY);
+    String description = (String) placeGuideEntity.getProperty(DESCRIPTION_PROPERTY);
+    long length = (long) placeGuideEntity.getProperty(LENGTH_PROPERTY);
+    String imageKey = (String) placeGuideEntity.getProperty(IMAGE_KEY_PROPERTY);
+    PlaceGuide placeGuide = 
+            new PlaceGuide.Builder(id, name, audioKey, creatorId, coordinate)
```

Comment:
no need that many spaces. Continue from previous line, 4 is standard

Reply:
fixed.